executor/src/witgen/machines/profiling.rs

```diff
@@ -81,7 +81,6 @@ pub fn reset_and_print_profile_summary() {
             // Update the call stack.
             match event {
                 Event::Start => {
-                    assert!(current_machine_id != id, "Unexpected recursive call!");
```

Comment:
This happens if witjitgen calls into another machine to witjitgen that one. Actually let me solve this in a different way.